### PR TITLE
[init] Get rid of some ENABLE_WALLET

### PR DESCRIPTION
**Pull request by maflcko**
c.f. second point of #7965

### CODE REVIEW

**Review by laanwj — APPROVED** [review]
nice
utACK https://github.com/bitcoin/bitcoin/pull/8760/commits/faddd62518c7b81c56ea2c9936cf850878c03e0b

**Review by dcousens — APPROVED** [review]
utACK

File: src/init.cpp [hunk 1]
```diff
@@ -759,10 +759,7 @@ void InitParameterInteraction()
     if (GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY)) {
         if (SoftSetBoolArg("-whitelistrelay", false))
             LogPrintf("%s: parameter interaction: -blocksonly=1 -> setting -whitelistrelay=0\n", __func__);
-#ifdef ENABLE_WALLET
-        if (SoftSetBoolArg("-walletbroadcast", false))
-            LogPrintf("%s: parameter interaction: -blocksonly=1 -> setting -walletbroadcast=0\n", __func__);
-#endif
+        // walletbroadcast is disabled in CWallet::ParameterInteraction()
```
Comment (jonasschnelli) [inline src/init.cpp]: nit: I think we can remove this comment.

**Review by jonasschnelli — APPROVED** [review]
utACK

File: src/init.cpp (hunk 1 above)
Comment (MarcoFalke) [inline src/init.cpp]: If people agree, I will just do it in my next wallet-init pull, so I don't invalidate the existing review here.

Comment (jonasschnelli) [inline src/init.cpp]: Fair enough.

**Comment by paveljanik** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/8760/commits/faddd62518c7b81c56ea2c9936cf850878c03e0b

File: src/init.cpp (hunk 1 above)
Comment (laanwj) [inline src/init.cpp]: I think it's fine to have this comment here. It was useful in reviewing at least, and maybe also for people that merge code that changes thes lines.

**Pull request merged**